src/java/utils/JsonHelper.java

```diff
@@ -14,4 +17,39 @@ public static JSONObject buildNoneLanguageMap(String label)
         languageMap.put("none", noneMap);
         return languageMap;
     }
+
+    /**
+    *  Make a hashmap with old @type from v2 API being keys and new type and format fields being values
+     *  to use in buildAnnotationResource
+    * */
+    private static HashMap buildBodyMap()
+    {
+        HashMap<String, JSONObject> map = new HashMap();
+        
+        JSONObject textMap = new JSONObject();
+        textMap.put("type", "TextualBody");
+        textMap.put("format", "text/plain");
+        map.put("cnt:ContentAsText", textMap);
```

Comment:
https://www.w3.org/TR/annotation-model/#embedded-textual-body Look at `TextualBody` and `bodyValue` and see if there is a discussion to have here about what would be the most "Presi 3" way to do this. We can all chat about it as well, but I would like to see what you have to say. In fact, if OSS sets up a meeting with IIIF folks, I know they will want to say something here about `purpose` and `motivation` as well.

Reply:
> Look at `TextualBody` and `bodyValue` and see if there is a discussion to have here about what would be the most "Presi 3" way to do this.

It says that `TextualBody` is preferred, since `bodyValue` does not store information about language and format.

Reply:
> I would say we need to change this for sure map.put("cnt:ContentAsText", textMap);. It doesn't seem right.

The way I thought about it was that we could get values for `type` and `format` properties based on old `@type`, not just specifying them as parameters to `buildAnnotationBody`. So right now `@type: "cnt:ContentAsText"` would give `{"type":"TextualBody","format":"text/plain"}`  because that's the value for `cnt:ContentAsText` key from `formatTextualBody` map. If someone wants to add support for images, for example, that person would then modify `formatTextualBody`, adding a value of `{"type":"Image","format": "image/jpeg"}` (from https://iiif.io/api/presentation/3.0/#57-content-resources) for `dctypes:Image` key. That really was the point of this whole helper function, as otherwise it would be just a bunch of `body.put(some key in body, some argument this function takes);` statements. 
But yes, the fact that `TextualBody` type can have a format of `text/plain` and `text/html` (and probably some other formats) is a clear problem.

Reply:
With multiple levels of "type" you will have to be careful what "type" you are detecting.  

Presentation 2 is
```
{
   "@type": "oa:Annotation",
   "motivation": "oad:transcribing",
   "resource": {
      "@type": "cnt:ContentAsText",
      "cnt:chars": "This is the second line"
   }
}
```

Presentation 3 is
```
{
    "type": "Annotation",
    "motivation": "transcribing",
    "body":{
        "type" : "TextualBody",
        "value" : "j'adore !",
        "format" : "text/plain",
        "language" : ["none"]
    }
 }
 ````

Notice there is no `cnt:ContentAsText` anymore.  It will always be that the whole Annotation needs converting, not just the `body` or just the `resource`.  

Your function will `map.put("cnt:ContentAsText", textMap)` which means buildBodyMap() is returning something like
```
{
   "cnt:ContentAsText":{
         "type": "TextualBody",
         "format": "text/plain"
   }
}
```
 Which doesn't match the Presentation 2 or Presentation 3 pattern.

Reply:
One buildBodyMap() returns that, but we then call `get()` on that getting only the value

`JSONObject body = (JSONObject) formatTextualBody().get(type);`

body will be 

'
'

Reply:
ok ok I see what you are going for there.  I expected `formatTextualBody()` to return what you needed, not something to read further to get what you needed.  I would expect it to return

```
"resource": {
      "@type": "cnt:ContentAsText",
      "cnt:chars": "This is the second line"
   }
```
For a prezi 2 Annotation "body" or

```
"body":{
        "type" : "TextualBody",
        "value" : "j'adore !",
        "format" : "text/plain",
        "language" : ["none"]
    }
```
For a prezi 3 Annotation "body"

None of functions format an object like
`{body:{}}`

Are you expecting upstream code to do the following to discern which body to use?

``` java
   JSONObject annoObject = new JSONObject();
   JSONObject bodyObject = new JSONObject();
  
   if(prezi3){
        bodyObject = buildAnnotationBody("cnt:ContentAsText", "some string value");
        annoObject.put("motivation", "transcribing");
        annoObject.put("body", bodyObject);
   }
   else{
       annoObject.put("motivation", "oad:transcribing");
       annoObject.put("resource", buildAnnotationBody("cnt:ContentAsText", "some string value"));
   }
   ...
```

Reply:
Ok, that makes sense.  Content negotiation may have you change it up, but at this time this is for "presentation 2 in presentation 3 out".
  
Make sure the code upstream knows `resource` is replaced with `body` if you aren't going to do that here.